### PR TITLE
support large inputs using chunking

### PR DESCRIPTION
**Pull request by artfulhacker**
should address #1

### CODE REVIEW

File: text.js [hunk 1]
```diff
@@ -178,7 +180,15 @@ FastTextDecoder.prototype.decode = function(buffer, options={stream: false}) {
     }
   }
 
-  return String.fromCharCode.apply(null, out);
+  // itterate over the out buffer
```
Comment (vapier) [inline text.js]: "iterate"

the comment should also explain why it's chunking things up (to avoid stack recursion)

File: text.js [hunk 2]
```diff
@@ -141,8 +141,10 @@ FastTextDecoder.prototype.decode = function(buffer, options={stream: false}) {
 
   const bytes = new Uint8Array(buffer);
   let pos = 0;
+  let chunksize = 0xFFFF;
   const len = bytes.length;
   const out = [];
+  let chunks = [];
```
Comment (vapier) [inline text.js]: use `const` instead

File: text.js [hunk 3]
```diff
@@ -178,7 +180,15 @@ FastTextDecoder.prototype.decode = function(buffer, options={stream: false}) {
     }
   }
 
-  return String.fromCharCode.apply(null, out);
+  // itterate over the out buffer
+	for (let i = 0; i < len; i+=chunksize)
```
Comment (vapier) [inline text.js]: incorrect indentation

needs spaces around the +=

File: text.js [hunk 4]
```diff
@@ -178,7 +180,15 @@ FastTextDecoder.prototype.decode = function(buffer, options={stream: false}) {
     }
   }
 
-  return String.fromCharCode.apply(null, out);
+  // itterate over the out buffer
+	for (let i = 0; i < len; i+=chunksize) 
+	{
```
Comment (vapier) [inline text.js]: this should be cuddled up to the previous line

File: text.js [hunk 5]
```diff
@@ -178,7 +180,15 @@ FastTextDecoder.prototype.decode = function(buffer, options={stream: false}) {
...
+		// slice chunks to charcodes and apply
+		chunks.push(String.fromCharCode.apply(null, out.slice(i, i+chunksize)));
```
Comment (vapier) [inline text.js]: space around the `+`

File: text.js [hunk 6]
```diff
@@ -178,7 +180,15 @@ FastTextDecoder.prototype.decode = function(buffer, options={stream: false}) {
...
+	}
+
+	// concat the chunks
+	return chunks.join("");
```
Comment (vapier) [inline text.js]: use single quotes

**Comment by samthor** [comment]
Thanks, but I've updated the codebase a different way.

**Pull request closed without merging**